src/meta/server_state.cpp

```diff
@@ -2864,5 +2883,79 @@ void server_state::clear_app_envs(const app_env_rpc &env_rpc)
                    new_envs.c_str());
         });
 }
+
+namespace {
+
+bool validate_target_max_replica_count_internal(int32_t max_replica_count,
+                                                int32_t alive_node_count,
+                                                std::string *hint_message)
+{
+    dassert_f(FLAGS_min_allowed_replica_count <= FLAGS_max_allowed_replica_count,
+              "min allowed replica count({}) should be <= max allowed replica count({})",
+              FLAGS_min_allowed_replica_count,
+              FLAGS_max_allowed_replica_count);
+
+    if (alive_node_count < 1) {
+        *hint_message = "all replica servers are unalive";
+        return false;
+    }
+
+    if (max_replica_count > FLAGS_max_allowed_replica_count) {
+        *hint_message = fmt::format("requested replica count({}) exceeds the max "
+                                    "allowed replica count({})",
+                                    max_replica_count,
+                                    FLAGS_max_allowed_replica_count);
+        return false;
+    }
+
+    if (max_replica_count < FLAGS_min_allowed_replica_count) {
+        *hint_message = fmt::format("requested replica count({}) is less than the min "
+                                    "allowed replica count({})",
+                                    max_replica_count,
+                                    FLAGS_min_allowed_replica_count);
+        return false;
+    }
+
+    if (max_replica_count > alive_node_count) {
+        *hint_message = fmt::format("there are not enough alive replica servers({}) "
+                                    "for the requested replica count({})",
+                                    alive_node_count,
+                                    max_replica_count);
+        return false;
+    }
+
+    return true;
+}
+
+} // anonymous namespace
+
+bool server_state::validate_target_max_replica_count(int32_t max_replica_count)
+{
+    std::string hint_message;
+
+    zauto_read_lock l(_lock);
+
+    auto alive_node_count = get_alive_node_count_unlocked();
+
+    bool is_valid = validate_target_max_replica_count_internal(
+        max_replica_count, alive_node_count, &hint_message);
+    if (!is_valid) {
+        dwarn_f("target max replica count is invalid: message={}", hint_message);
+    }
+
+    return is_valid;
+}
+
+int32_t server_state::get_alive_node_count_unlocked()
```

Comment:
meta_service has a local variety called `_alive_set`, you can get alive node count through it.
https://github.com/XiaoMi/rdsn/blob/289eb4609ae781e3bcc6bbc6bfeae64dfd8fa785/src/meta/meta_service.h#L269